### ISSUE TITLE
Projectproperties won't sync to hub if entity for the project is missing in the 'Prosjekter' list

### ISSUE THREAD

**Issue by Remi749** [Puzzlepart/prosjektportalen365#715]
**Describe the bug**
Projectproperties won't sync to hub if entity for the project is missing in the 'Prosjekter' list.

...\ProjectWebParts\src\data\SPDataAdapter\index.ts

`await this.entityService.updateEntityItem(this.settings.siteId, properties)`

Possible solution:
Use entityservice to check if there exists an entityItem for the project, if not, create.

![image](https://user-images.githubusercontent.com/28678468/171852874-76c768e7-fed1-4b51-8074-390adaeed766.png)

Screenshot of error in console:
![image](https://user-images.githubusercontent.com/28678468/171852728-cceb8ff8-de06-4abe-9ae7-43611b80ac1b.png)

**To Reproduce**

1. Go to a project that does not have an element in the 'Prosjekter' list, remove the element if there is any (to test)
2. Go to the 'Rediger egenskaper' in the 'Prosjektinformasjon' webpart
3. Make some changes or just press 'Lagre'

**Expected behavior**
Properties should be synced, entityItem should be created if it doesn't exist so that the updateEntityItem function can run properly

**Comment by okms** [comment]
Fix is to create element if it does not exist

**Comment by olemp** [comment]
An element is created when the project is provisioned. In what case will there be no item in the "Prosjekter" list @okms @Remi749?

**Comment by tonnesr** [comment]
Issue is currently blocked by issues with CI.

**Comment by olemp** [comment]
> Issue is currently blocked by issues with CI.

Not sure what we can do there. I tried deleting some artefacts from previous runs (and the logs), but didn't make a difference.

![image](https://user-images.githubusercontent.com/7606007/194557443-368fb992-0961-481b-a64a-5540f1530635.png)

**Comment by olemp** [comment]
> > Issue is currently blocked by issues with CI.
> 
> Not sure what we can do there. I tried deleting some artefacts from previous runs (and the logs), but didn't make a difference.
> 
> ![image](https://user-images.githubusercontent.com/7606007/194557443-368fb992-0961-481b-a64a-5540f1530635.png)

fyi @tonnesr @Remi749

**Comment by olemp** [comment]
I think this is a edge case. We can't safeguard ourselves against all kinds of deletion.

**Comment by Remi749** [comment]
Should we instead give users a little error message that the project is missing incase this happens?

**Comment by olemp** [comment]
> Should we instead give users a little error message that the project is missing incase this happens?

We could, but users can also

- Delete the front page
- Delete navigation elements
- Delete all items in _Maloppsett_
- Remove the site design

It's nice to handle edge cases, but where do we set the limit?

**Comment by Remi749** [comment]
> > Should we instead give users a little error message that the project is missing incase this happens?
> 
> We could, but users can also
> 
> * Delete the front page
> * Delete navigation elements
> * Delete all items in _Maloppsett_
> * Remove the site design
> 
> It's nice to handle edge cases, but where do we set the limit?

Valid point. Let this be a won't fix.
